cairo/src/error.rs

```diff
@@ -1,143 +1,156 @@
     #[doc(alias = "STATUS_INVALID_CONTENT")]
     InvalidContent,
-    #[error("Invalid Format")]
     #[doc(alias = "STATUS_INVALID_FORMAT")]
     InvalidFormat,
-    #[error("Invalid Visual")]
     #[doc(alias = "STATUS_INVALID_VISUAL")]
     InvalidVisual,
-    #[error("File Not Found")]
     #[doc(alias = "STATUS_FILE_NOT_FOUND")]
     FileNotFound,
-    #[error("Invalid Dash")]
     #[doc(alias = "STATUS_INVALID_DASH")]
     InvalidDash,
-    #[error("Invalid Dash Comment")]
     #[doc(alias = "STATUS_INVALID_DSC_COMMENT")]
     InvalidDscComment,
-    #[error("Invalid Index")]
     #[doc(alias = "STATUS_INVALID_INDEX")]
     InvalidIndex,
-    #[error("Clip Not Representable")]
     #[doc(alias = "STATUS_CLIP_NOT_REPRESENTABLE")]
     ClipNotRepresentable,
-    #[error("Temp File Error")]
     #[doc(alias = "STATUS_TEMP_FILE_ERROR")]
     TempFileError,
-    #[error("Invalid Stride")]
     #[doc(alias = "STATUS_INVALID_STRIDE")]
     InvalidStride,
-    #[error("Font Type Mismatch")]
     #[doc(alias = "STATUS_FONT_TYPE_MISMATCH")]
     FontTypeMismatch,
-    #[error("User Font Immutable")]
     #[doc(alias = "STATUS_USER_FONT_IMMUTABLE")]
     UserFontImmutable,
-    #[error("User Font Error")]
     #[doc(alias = "STATUS_USER_FONT_ERROR")]
     UserFontError,
-    #[error("Negative Count")]
     #[doc(alias = "STATUS_NEGATIVE_COUNT")]
     NegativeCount,
-    #[error("Invalid Clusters")]
     #[doc(alias = "STATUS_INVALID_CLUSTERS")]
     InvalidClusters,
-    #[error("Invalid Slant")]
     #[doc(alias = "STATUS_INVALID_SLANT")]
     InvalidSlant,
-    #[error("Invalid Weight")]
     #[doc(alias = "STATUS_INVALID_WEIGHT")]
     InvalidWeight,
-    #[error("Invalid Size")]
     #[doc(alias = "STATUS_INVALID_SIZE")]
     InvalidSize,
-    #[error("User Font Not Implemented")]
     #[doc(alias = "STATUS_USER_FONT_NOT_IMPLEMENTED")]
     UserFontNotImplemented,
-    #[error("Device Type Mismatch")]
     #[doc(alias = "STATUS_DEVICE_TYPE_MISMATCH")]
     DeviceTypeMismatch,
-    #[error("Device Error")]
     #[doc(alias = "STATUS_DEVICE_ERROR")]
     DeviceError,
-    #[error("Invalid Mesh Construction")]
     #[doc(alias = "STATUS_INVALID_MESH_CONSTRUCTION")]
     InvalidMeshConstruction,
-    #[error("Device Finished")]
     #[doc(alias = "STATUS_DEVICE_FINISHED")]
     DeviceFinished,
-    #[error("JBig2Global Missing")]
     #[doc(alias = "STATUS_J_BIG2_GLOBAL_MISSING")]
     JBig2GlobalMissing,
-    #[error("PNG Error")]
     #[doc(alias = "STATUS_PNG_ERROR")]
     PngError,
-    #[error("Freetype Error")]
     #[doc(alias = "STATUS_FREETYPE_ERROR")]
     FreetypeError,
-    #[error("Win32Gdi Error")]
     #[doc(alias = "STATUS_WIN32_GDI_ERROR")]
     Win32GdiError,
-    #[error("LastStatus")]
     #[doc(alias = "STATUS_LAST_STATUS")]
     LastStatus,
-    #[error("Unknown {0}")]
     #[doc(hidden)]
     __Unknown(i32),
 }
+
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
```

Comment:
This is the default implementation already

Reply:
No, this is needed, or I misunderstood what you meant.

Reply:
You don't have to provide the source method if you're going to return None. That's the default implementation

Reply:
Apparently implementing `Error` is complicated enough that you missed this once when implementing it and a second time after I pointed it out :shrug:

Reply:
Sorry...